src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -317,87 +358,42 @@ public void verifyToken() throws LoginException
     // @param alreadyFailed If has already been a failed attempt with the current configuration
     public void verifyToken(boolean alreadyFailed) throws LoginException
     {
-
-        RestActionImpl<DataObject> login = new RestActionImpl<DataObject>(this, Route.Self.GET_SELF.compile())
-        {
-            @Override
-            public void handleResponse(Response response, Request<DataObject> request)
-            {
-                if (response.isOk())
-                    request.onSuccess(response.getObject());
-                else if (response.isRateLimit())
-                    request.onFailure(new RateLimitedException(request.getRoute(), response.retryAfter));
-                else if (response.code == 401)
-                    request.onSuccess(null);
-                else
-                    request.onFailure(new LoginException("When verifying the authenticity of the provided token, Discord returned an unknown response:\n" +
-                        response.toString()));
-            }
-        };
-
-        DataObject userResponse;
-
         if (!alreadyFailed)
```

Comment:
That would duplicate code, I don't think that's worth it.

Reply:
Since nothing actually calls `verifyToken(boolean)` except for the `verifyToken()` method, can we instead remove the `verifyToken(boolean)` variant completely? Afaik this was only here to handle the check to see if an incorrect AccountType had been provided for a valid token.

This would _also_ remove the if-statement scoping.

Reply:
Probably could also move the contents of `checkToken` back into `verifyToken` as we only had this dual-method concept so that we could reuse the `login` RestAction instead of constructing it multiple times.

Reply:
It is also called by the SessionControllerAdapter to throw if the gateway/bot endpoint returns a 401:

https://github.com/DV8FromTheWorld/JDA/blob/b8a716d3152112e8f8cd7f05fc3fe8353871de83/src/main/java/net/dv8tion/jda/api/utils/SessionControllerAdapter.java#L112

Reply:
That definitely feels like a misuse of code. It is calling a complex system instead of just doing
```
shutdownNow
throw new LoginException
```